### PR TITLE
HBHE: add M2-chi2 into the HBHErecHit data formats

### PR DESCRIPTION
**Pull request by mariadalfonso**
Added the Method 2-chisquare into the HBHERecHit data formats.
@igv4321

### CODE REVIEW

**Comment by slava77** [comment]
@mariadalfonso 
is it really just for M2?
The name is generic enough to support other use cases.
Is there an equivalent of chi2 in M3?

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by mariadalfonso** [comment]
@slava77 
M3 doens't have a meaningful value @jaylawhorn

**Comment by slava77** [comment]
@mariadalfonso @igv4321
is chi2 enough for your needs?
Should ndf be added as well? (we have one and 3-pulse fit versions active now; unclear if 2-pulse can show up in the future or some shape parameter be added. So, e.g. chi2~10 may be on a high side for 1-pulse, but kind of OK for 3-pulse).

Also, as more values are added, to the event content, the cost on disk increases, even if it is with a shorter life time.
Consider packing this as some e.g int/255, which should be well compressable, compared to typically worse than 2 compression factor for floats.

File: DataFormats/HcalRecHit/interface/HBHERecHit.h [hunk 1]
```diff
@@ -22,6 +22,9 @@ class HBHERecHit : public CaloRecHit {
   /// get the id
   inline HcalDetId id() const { return HcalDetId(detid()); }
 
+  inline void setChiSquared(const float chi2) {chiSquared_ = chi2;}
+  inline float getChiSquared() const {return chiSquared_;}
```
Comment (slava77) [inline DataFormats/HcalRecHit/interface/HBHERecHit.h]: please change to ```chiSquared``` to preserve the style with all other access methods, which do not have a "get" prefix.

Comment (slava77) [inline DataFormats/HcalRecHit/interface/HBHERecHit.h]: ... better yet, use chi2 in the method names to match what we have in track and ecal hit classes.

Comment (mariadalfonso) [inline DataFormats/HcalRecHit/interface/HBHERecHit.h]: @slava77 renamed getChiSquared into chi2

**Comment by mariadalfonso** [comment]
@slava77 The ndof is not needed at the moment. 
There is a flag in the rechit to distinguish 1 and 3-pulse fits, and in the future we can store the reduced chi2.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
chi2 values are now filled
(color code is inverted wrt the legend: red is baseline and black is with this PR)

Looking at data
2016 single photon wf 136.731
all_origvssign771_runsingleph2016bwf136p731c_log10hbherechitssorted_hbhereco__rereco_obj_obj_chi2
about 15% have a default value (999).
MC with pileup has a similar distribution.

2017 HCALdev wf has a larger fraction at default value
all_origvssign771_ttbar13tev2017hcaldevwf10624p0c_log10hbherechitssorted_hbhereco__reco_obj_obj_chi2

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/15916 be23969
- code change is in line with the PR description and the follow up review
- jenkins tests pass and comparisons show no differences (the new variable is not monitored)
- local tests with added monitoring of chi2 shows that the variable is now filled

**Comment by davidlange6** [comment]
+1

**Pull request merged**